src/ctia/lib/utils.clj

```diff
@@ -134,3 +134,38 @@
 (def deep-remove-nils
   "Remove nil values from a deep nested map recursively"
   (partial deep-filter some?))
+
+(defn service-subgraph
+  "Returns a subgraph of a Trapperkeeper service graph, omitting
+  any missing entries.
+  
+  (select-services
+    {:ConfigService {:get-in-config <...>
+                     :get-config <...>}
+     :FooService {:f1 <...>
+                  :f2 <...>}
+     :BarService {:b1 <...>}}
+    :ConfigService [:get-config]
+    :FooService [:f2 :f3]
+    :MissingService [:m1])
+  ;=> {:ConfigService {:get-config <...>}
+  ;    :FooService {:f2 <...>}}
+  "
+  [graph & selectors]
+  {:pre [(map? graph)]}
+  (reduce (fn [out [service-kw & [fn-kws :as is-even]]]
+            (assert is-even
+                    (str "Uneven number of selectors: "
+                         (count selectors)))
```

Comment:
I'll add comments in the code. Since we're reducing over a `(partition-all 2)` each partition is either size 1 or 2. All but maybe the last one will be size 2.

This avoids `count` which would retraverse sequences.

FWIW there are tests that ensure this works, which is what enabled me to refactor this away from `count` more confidently via TDD.